### PR TITLE
Add LocalizeStaticMixin and update d2l-dialog to use it

### PR DESCRIPTION
**Pull request by dbatiste**
* introduces new `LocalizeStaticMixin` to avoid repeated code
* rejig readme to include the new mixin
* update d2l-dialog to use it

### CODE REVIEW

File: mixins/localize-static-mixin.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { LocalizeMixin } from './localize-mixin.js';
+
+export const LocalizeStaticMixin = superclass => class extends LocalizeMixin(superclass) {
+
+	static async getLocalizeResources(langs) {
+		langs.forEach((lang) => {
```
Comment (dbatiste) [inline mixins/localize-static-mixin.js]: This is just the current logic that's used.  I can update the logic to do something similar to what @mpharoah-d2l did for a more granular fallback, but I wanted to open the PR to get feedback on the idea of the static/local mixin approach.

File: mixins/localize-static-mixin.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { LocalizeMixin } from './localize-mixin.js';
+
+export const LocalizeStaticMixin = superclass => class extends LocalizeMixin(superclass) {
+
```
Comment (dlockhart) [inline mixins/localize-static-mixin.js]: Should this define the static resources property and default it to return `{ 'en': {} }` (to avoid explosions)?

Comment (dbatiste) [reply mixins/localize-static-mixin.js]: Yeah good point... it will indeed explode otherwise.

File: mixins/localize-mixin.md [hunk 3]
```diff
@@ -240,4 +229,4 @@ time.formatFileSize(1234567.89);
 
 ## Automatic Language, Timezone and Override Support
 
-The user's language, timezone and any D2L locale overrides are automatically fetched by `LocalizeMixin` from the `<html>` element's `lang`, `data-timezone` and `data-intl-overrides` attributes respectively.
+The user's language, timezone and any D2L locale overrides are automatically fetched by he localization mixins from the `<html>` element's `lang`, `data-timezone` and `data-intl-overrides` attributes respectively.
```
Comment (dlockhart) [inline mixins/localize-mixin.md]: Typo: "fetched by he localization mixins..."

File: components/dialog/dialog.js [hunk 4]
```diff
@@ -56,6 +57,14 @@ class Dialog extends DialogMixin(LitElement) {
 		`];
 	}
 
+	static get resources() {
+		return {
+			'en': {
+				close: 'Close this dialog'
```
Comment (dlockhart) [inline components/dialog/dialog.js]: We probably have this lang term translated already in the LMS? Could probably snag all the translations...

File: mixins/localize-static-mixin.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+		const resolvedResources = Object.assign({}, this.resources[fallbackLang]);
+
+		langs.reverse().forEach((lang) => {
+			if (this.resources[lang]) {
+				resolvedLang = lang;
+				Object.assign(resolvedResources, this.resources[lang]);
+			}
+		});
+
+		return {
```
Comment (dbatiste) [inline mixins/localize-static-mixin.js]: I've adapted/copied the logic.  The merge is pessimistic, in the sense that it iterates and merges all the languages even if it didn't need to.  Might not be a big deal for small sets.  I think this code will get called for every instance of a component, and it seems like we could cache the resolved resources so we only do this work once.

File: components/dialog/dialog.js (hunk 4 above)
Comment (dbatiste) [reply components/dialog/dialog.js]: I'll take a look!

File: mixins/localize-static-mixin.js (hunk 5 above)
Comment (dlockhart) [reply mixins/localize-static-mixin.js]: Agreed -- I created #232 to track that.

**Pull request merged**